neps/nep-0539.md

```diff
@@ -377,7 +373,39 @@ https://github.com/near/nearcore/pull/10918
 Here are the most important details which are not already described in the
 specification above but are defined in the reference implementation.
 
-### Changes to Chunk Extra
+### Efficiently computing the congestion information
+
+The congestion information is computed based on the gas and size of the incoming
+queue and the outgoing buffers. A naive implementation would just iterate over all
+of the receipts in the queue and buffers and sum up the relevant metrics. This
+approach is slow and, in the context of stateless validation, would add too much 
+to the state witness size. In order to prevent those issues we consider two 
+alternative optimizations. Both use the same principle of caching the previously
+calculated metrics and updating them based on the changes to the incoming queue 
+and outgoing buffers. 
+
+#### Option 1 - store the total size and total gas in the state
+
+In this option we store the total gas and total size of the incoming queue and 
+the outgoing receipts in the state along the respective queue or buffers. Those
+values will be updated as receipts are added or removed from the queue. 
+
+In this case the CongestionInfo struct can remain small and only reflect the 
+information needed by other shards. 
+
+```rust
+pub struct CongestionInfo {
+  congestion_level: u8,
+  allowed_shard: u64,
```

Comment:
Nit: Can we change the order of the fields? The misaligned `u64` hurts my eyes when I look at it^^

Reply:
I copied the CongestionInfo from the SmoothTrafficLight congestion model, are we aiming for something different?

Reply:
Also I think u32 for shard id should be more than enough

Reply:
But I'd rather do it in a separate PR

Reply:
Ah, yeah you are right. I got things mixed up because the reference implementation has a bug in `shard_accepts_transactions` ^^

Reply:
cool, I'll fix that quickly